internal/staticanalysis/result.go

```diff
@@ -49,3 +41,44 @@ func (r SingleResult) String() string {
 
 	return strings.Join(parts, "\n\n")
 }
+
+// ProduceSerializableResult converts the data in this Result object into the
+// public staticanalysis.Results format defined in pkg/api/staticanalysis.
+func (r *Result) ProduceSerializableResult() *staticanalysis.Results {
```

Comment:
nit: i wonder if `ToSerializableResult` or `ToAPIResult` is a slightly better name.

"Produce" is a bit long, and it is common for `FromFormat`, `ToFormat` to be used.

Reply:
good point, I'll make the change